### PR TITLE
add more structure to support STP Transport Layer for MIPI SyS-T

### PR DESCRIPTION
**Pull request by wentongwu**
Add more structure to support STP Transport Layer for MIPI SyS-T.

Signed-off-by: Wentong Wu <wentong.wu@intel.com>

### CODE REVIEW

**Comment by wentongwu** [comment]
@nashif could you please help review this PR so that we can move forward for https://github.com/zephyrproject-rtos/zephyr/pull/28717? thanks

File: include/mipi_syst/platform.h [hunk 1]
```diff
@@ -13,18 +13,38 @@
 extern "C" {
```
Comment (nashif) [inline include/mipi_syst/platform.h:13]: was this file added by you? if yes, it should not be here, put it under zephyr to distinguish it from the upstream code.

**Review by nashif — CHANGES_REQUESTED** [review]
Add more detailed commit message and a body to the commit message just like we do to core Zephyr tree.

File: include/mipi_syst/platform.h (hunk 1 above)
Comment (wentongwu) [reply include/mipi_syst/platform.h:13]: @nashif the file(library/include/mipi_syst.h.in) in syst library will include the platform.h in this way `#include "mipi_syst/platform.h"`, IMHO it's not public API for logging sub system, so it's not reasonable put it under include/logging; also there is no include folder under subsys/logging, and I'm not sure if adding include folder under subsys/logging is acceptable or not. And putting it here, just don't want to break anything for the logging system.
But anyway if it's expected, I can move this to Zephyr tree.

Comment (nashif) [reply include/mipi_syst/platform.h:13]: see #3

**Comment by wentongwu** [comment]
close this one, will add the changes when platform.h merged to Zephyr tree.

**Pull request closed without merging**